src/hotspot/share/memory/metaspace.hpp

```diff
@@ -124,6 +124,10 @@ class Metaspace : public AllStatic {
   static MetaWord* allocate(ClassLoaderData* loader_data, size_t word_size,
                             MetaspaceObj::Type type, TRAPS);
 
+  // Nothrow version of allocate which can be called by a non-Java thread.
```

Comment:
"Nothrow" is a C++ concept. I would just say "Non-TRAPS version ... Returns NULL on failure."

Reply:
Yes, true.  Particularly since it had to be declared with throws().

Reply:
got it!